lms/djangoapps/courseware/grades.py

```diff
@@ -499,10 +559,13 @@ def _progress_summary(student, course, course_structure=None):
         course_structure = get_course_blocks(student, course.location)
     if not len(course_structure):
         return None
-    scorable_locations = [block_key for block_key in course_structure if possibly_scored(block_key)]
-
-    with outer_atomic():
-        scores_client = ScoresClient.create_for_locations(course.id, student.id, scorable_locations)
+    if scores_client is None:
+        if grading_context_result is None:
+            scorable_locations = [block_key for block_key in course_structure if possibly_scored(block_key)]
+        else:  # TODO: these two list are equivalent, yeah?
```

Comment:
Looking at the latest master, it would represent about a 10% reduction. The overall effect is probably larger than that, since it doesn't include the speedup from the transform traversal PR.

Do we need additional filtering? The pattern is just "grab all the scores and then get() them one by one from that pool as you need", no?